src/versions.jl

```diff
@@ -251,12 +255,12 @@ end
 Base.intersect(a::VersionNumber, B::VersionSpec) = a in B ? VersionSpec(a) : empty_versionspec
 Base.intersect(A::VersionSpec, b::VersionNumber) = intersect(b, A)
 
-Base.union(A::VersionSpec, B::VersionSpec) = union!(copy(A), B)
-function Base.union!(A::VersionSpec, B::VersionSpec)
```

Comment:
The point of this is to consider a `VersionSpec` immutable so that we can use multiple instances of the same `VersionSpec` and not have to create tens of thousands of `VersionSpec("1")` just in case it get mutated somewhere.